framework/src/base/FEProblem.C

```diff
@@ -704,16 +709,47 @@ FEProblem::getMaxScalarOrder() const
 }
 
 void
+FEProblem::checkNonlocalCoupling()
```

Comment:
But Kernels could be switched on and off...?

On Wed, Jul 13, 2016, 12:14 PM Derek Gaston <notifications@github.com>
wrote:

> In framework/src/base/FEProblem.C
> <https://github.com/idaholab/moose/pull/7216#discussion_r70679173>:
>
> > @@ -704,16 +709,47 @@ FEProblem::getMaxScalarOrder() const
> >  }
> >
> >  void
> > +FEProblem::checkNonlocalCoupling()
>
> This looks really expensive... even if it's only happening once per
> timestep... n_vars*n_subdomains could be HUGE for some problems (like
> neutronics).
>
> Somehow you need to do this up front and cache the data...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/7216/files/a942428563b3925e603fcd4fe78ca0a438e12aa4#r70679173>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMWPhpFZgMcHI-mgbVMWLZ28gIrZmx1ks5qVSsNgaJpZM4I6DA4>
> .
>